### ISSUE TITLE
[QST] Generate predictions without using TIS

### ISSUE THREAD

**Issue by IamGianluca** [NVIDIA-Merlin/Transformers4Rec#372]
Is it possible to generate predictions directly from the `Trainer`? I was hoping, for rapid development purposes, to load my validation/test set, apply the NVTabular workflow and pass it to something like `trainer.predict_step()` to generate the best N recommendations for each session included in the validation/test set. However, I can't find any example in the documentation of such functionality.

The `Trainer.prediction_step()` method doesn't take an `nvt.Dataset` or `Dataloader` and instead expects the `input` to be a `Dict[str, torch.Tensor]`. I'm wondering how do I get my test set in a state that can be passed to `Trainer.prediction_step()` in order to return the top N recommendations for each sequence?

**Comment by rnyak** [comment]
@IamGianluca you can simply do that:

```
batch = next(iter(trainer.get_eval_dataloader()))

model.eval()
model(batch, training=False)

# run the following to generate only predictions (logit values) for the given batch
model(batch, training=False)['predictions']

```
Let us know if that helps.

**Comment by IamGianluca** [comment]
Thank you @rnyak :+1:
